language/stdlib/modules/transaction_fee_distribution.mvir

```diff
@@ -0,0 +1,104 @@
+// This module implements a basic transaction fee distribution logic.
+//
+// We have made a couple design decisions here that are worth noting:
+//  * We pay out per-block as opposed to per-epoch. In the current distribution scheme (equal to
+//    all validators) there isn't any real difference between this and per-epoch distribution.
+//    However, other distribution schemes may have different economic implications depending upon
+//    if fees are distributed per-block or per-epoch.
+//  * In the handling of dust, it could be beneficial to be the first validator in the validating
```

Comment:
I don't think this comment doesn't make sense without additional context (e.g., that "dust" == `remainder` in the code). Also: didn't we agree with @aching that paying the reward forward into the pot for the next epoch was the preferred approach here?

Reply:
I thought we had agreed on that, but then changed our minds to instead pay it all to the validators in the current epoch. But if I mis-remembered, I can change it. My concern is that this would then lead to possibly different validators being paid out from a pot of money that should have been distributed to the validators in the previous epoch.

Reply:
I was mostly interested in getting rid of the `remainder` logic, which would:
- make the code smaller by at least 20-25 lines
- make it possible to verify that `destroy_zero` won't fail. (unfortunately, I think it's highly unlikely that the verifier will be able to infer the Tim-loop invariant because it involves divisibility)

> My concern is that this would then lead to possibly different validators being paid out from a pot of money that should have been distributed to the validators in the previous epoch.

I think this would be a very valid concern if (a) the amount in question was large (b) it were easy for future validators to game the system to ensure that they will get the payment instead of the current set. But I don't think either of these is true. For (a) The amount that the different validators will be paid out is at most 1 microlibra * num_validators (as your comment below indicates). The gas cost of the remainder logic is almost surely larger than this amount for any nonzero gas price and a realistic number of validators (e.g., 100).